ui/src/tool-bar/workspace-selection.tsx

```diff
@@ -13,6 +13,31 @@ import { WorkspaceDeleteConfirmDialog } from 'tool-bar/workspace-delete-confirm-
 import { LayoutContext } from 'common/common-model';
 import { Button, ButtonVariant } from 'vayla-design-lib/button/button';
 import { Icons } from 'vayla-design-lib/icon/Icon';
+import { useTrackLayoutAppSelector } from 'store/hooks';
+import { createDelegates } from 'store/store-utils';
+import { trackLayoutActionCreators } from 'track-layout/track-layout-slice';
+
+type WorkspaceSelectionContainerProps = {
+    setSelectingWorkspace: (selectingWorkspace: boolean) => void;
+    selectingWorkspace: boolean;
+};
+
+export const WorkspaceSelectionContainer: React.FC<WorkspaceSelectionContainerProps> = ({
+    setSelectingWorkspace,
+    selectingWorkspace,
+}) => {
+    const trackLayoutState = useTrackLayoutAppSelector((state) => state);
+    const delegates = React.useMemo(() => createDelegates(trackLayoutActionCreators), []);
```

Comment:
Tästä on ollut kahta käytäntöä, joko `useMemo()`:lla tai ilman, ja vaikka se onkin kätevä perffioptimointi useammin päivittyvissä komponenteissa, niin vaikuttaisi melko coin flipiltä että kumpaa mallia on käytetty näissä vähemmän päivittyvissä. Täällä tällä kertaa näin